src/extension/wps/src/main/java/au/org/emii/gogoduck/download/DownloadError.java

```diff
@@ -0,0 +1,21 @@
+package au.org.emii.gogoduck.download;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+
+/**
+ * Download error metadata
+ */
+public class DownloadError extends Download {
+    private final IOException exception;
+
```

Comment:
How about introducing new exception type DownloadException and using it here? The code will still work though.